src/ip/QosConfig.cc

```diff
@@ -465,64 +467,61 @@ Ip::Qos::Config::parseConfigLine()
  * which means no StoreEntry references. Just a basic char* buffer.
 */
 void
-Ip::Qos::Config::dumpConfigLine(char *entry, const char *name) const
+Ip::Qos::Config::dumpConfigLine(std::ostream &os, const char *name) const
 {
-    char *p = entry;
     if (isHitTosActive()) {
-
-        p += snprintf(p, 11, "%s", name); // strlen("qos_flows ");
-        p += snprintf(p, 4, "%s", "tos");
+        os << name << " tos";
 
         if (tosLocalHit > 0) {
-            p += snprintf(p, 16, " local-hit=0x%02X", tosLocalHit);
+            os << " local-hit=" << AsHex(tosLocalHit);
```

Comment:
For consistency sake, please use the lowercase asHex() wrapper rather than instantiating an object of the templated class AsHex. Same for other stream manipulators. IIRC, in some special cases, the compiler will not be able to guess the template type when instantiating an object, but will be able to guess it when calling a function. That is why we usually provide those wrappers...

Reply:
> I'll post a separate PR to add this snippet of knowledge to the IoManip.h file

Please note that it would be difficult to document this detail further without duplicating that documentation. We already have many such wrappers (at least two in IoManip.h and five more for various AsyncCalls: `git grep -E 'helper.*to (ease|simplify).*creation'`). We could hope that folks would not see any similar AsHex usage in the existing code and use asHex() instead, but that obviously does not really work well.

Alternatively, we can stop thriving for consistency in this area -- use AsHex() and asHex() interchangeably, without any change requests to rename.

A third option I can think of is to prove that asHex() is not needed -- that the cases such wrappers are meant to address do not apply to AsHex class. In that case, we can remove asHex() completely and solve the consistency problem that way!